### ISSUE TITLE
Markers without titles or snippets have empty InfoWindows

### ISSUE THREAD

**Issue by danswick** [mapbox/mapbox-gl-native#3559]
Tapping a marker without a `title` or `snippet` causes an empty infoWindow to pop up. I've seen this issue come through in at least two help requests and was able to replicate myself with:

``` java
mapView.addMarker(new MarkerOptions()
    .position(new LatLng(41.885, -87.679))
    .icon(icon));
```

Which results in (don't mind the incorrectly sized icon):

![image](https://cloud.githubusercontent.com/assets/2365503/12342429/206db202-badf-11e5-82b6-bb37180d5278.png)

Earlier versions of the Mapbox Android SDK produced a similar result with a slightly different style:

![image](https://cloud.githubusercontent.com/assets/2365503/12342407/f130d866-bade-11e5-9b6e-ae157f7b59cb.png)

**Comment by tobrun** [comment]
I will pick this up for release in #3537

**Comment by tobrun** [comment]
I will ignore the click event when there is no title and no snippet availlable

**Comment by tobrun** [comment]
Note that this fix will land on the release branch of 3.1.0 and not on master

**Comment by tobrun** [comment]
This has landed in e22b0270f486cbece252f7faec55cd927403200d in the release branch of 3.1.0 #3537

**Comment by danswick** [comment]
:boom: thanks for knocking this out so quickly, @tobrun!
